sdk/servicebus/Azure.Messaging.ServiceBus/tests/Processor/ProcessorLiveTests.cs

```diff
@@ -1534,10 +1545,16 @@ async Task ProcessMessage(ProcessMessageEventArgs args)
                     };
                     SimulateNetworkFailure(client);
                     await Task.Delay(lockDuration.Add(lockDuration));
-                    Assert.IsTrue(messageLockLostRaised);
-                    Assert.IsTrue(args.MessageLockCancellationToken.IsCancellationRequested);
-                    Assert.IsFalse(args.CancellationToken.IsCancellationRequested);
-                    tcs.SetResult(true);
+                    try
+                    {
+                        Assert.IsFalse(messageLockLostRaised);
+                        Assert.IsFalse(args.MessageLockCancellationToken.IsCancellationRequested);
+                        Assert.IsFalse(args.CancellationToken.IsCancellationRequested);
+                    }
+                    finally
```

Comment:
May be worth going through the other processor tests at some point to ensure we use this pattern to avoid timeouts for test failures. Just updated the newly added tests in this PR.